From: Elijah Wright <git@elijahs.space>
Subject: kernel: relay: use __GFP_ZERO in relay_alloc_buf
Date: Tue, 10 Jun 2025 15:56:28 -0700

Passing the __GFP_ZERO flag to alloc_page should result in less overhead
th= an using memset()

Link: https://lkml.kernel.org/r/20250610225639.314970-3-git@elijahs.space
Signed-off-by: Elijah Wright <git@elijahs.space>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 kernel/relay.c |    3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

--- a/kernel/relay.c~kernel-relay-use-__gfp_zero-in-relay_alloc_buf
+++ a/kernel/relay.c
@@ -118,7 +118,7 @@ static void *relay_alloc_buf(struct rcha
 		return NULL;
 
 	for (i = 0; i < n_pages; i++) {
-		buf->page_array[i] = alloc_page(GFP_KERNEL);
+		buf->page_array[i] = alloc_page(GFP_KERNEL | __GFP_ZERO);
 		if (unlikely(!buf->page_array[i]))
 			goto depopulate;
 		set_page_private(buf->page_array[i], (unsigned long)buf);
@@ -127,7 +127,6 @@ static void *relay_alloc_buf(struct rcha
 	if (!mem)
 		goto depopulate;
 
-	memset(mem, 0, *size);
 	buf->page_count = n_pages;
 	return mem;
 
_
